src/axom/spin/BVH.hpp

```diff
@@ -168,14 +177,11 @@ class BVH
    * \pre boxes != nullptr
    * \pre numItems > 0
    */
-  BVH(const FloatType* boxes,
-      IndexType numItems,
-      int allocatorID = axom::execution_space<ExecSpace>::allocatorID());
+  int initialize(const BoxType* boxes,
+                 IndexType numItems,
+                 int allocatorID = axom::execution_space<ExecSpace>::allocatorID());
```

Comment:
Design question: Given that BVH now has a member `allocator`, should `initialize` have a default value for `allocatorID`?

A different idea might be provide two overloads:
```cpp
BVH::initialize(const BoxType*, IndexType); // use the member allocator
BVH::initialize(const BoxType*, IndexType, int allocatorID); // overload, use the provided allocator instead
```
Where the first function would call the second using the member allocator.

The distinction would be if the user modified the `axom::allocatorID`, between calling the `BVH` constructor and calling `initialize`. I'm pretty sure that `axom::execution_space<ExecSpace>::allocatorID()` is a singleton.

Reply:
Good point: if `allocatorID` is a member field, it doesn't make much sense to set it to a default `allocatorID()` in the constructor, then do it again in `initialize()`.

How do you feel about removing the allocator argument from `initialize()` entirely, and making it an optional `setAllocatorID(int)` member function? All we really do with that argument is set `m_allocatorID = allocatorID`.

Reply:
That works for me.

Maybe add a note to the doxygen that it uses a previously set allocator in `setAllocatorID` or the default allocator?

Reply:
Done 👍